src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -355,7 +355,10 @@ internal PSKeyword GetLogKeywords()
             fileLock.EnterReadLock();
             try
             {
-                using (var streamReader = new StreamReader(fileName))
+                // The config file can be locked by another process
+                // so we wait some milliseconds in 'WaitForFile()' for recovery before stop current process.
+                using (var readerStream = WaitForFile(fileName, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
```

Comment:
With `FileShare.Read` the PowerShell Core process will _terminate_ if anybody open the file for writing. We aim to _avoid_ this and open the file with mininal lock level. Also we don't keep the file opened: every time we create new StreamReader on very short time. There still can be a race condition if we read partially changed file but it is another issue - what PowerShell Core should do if the config file is broken.

Reply:
IMO, this permission raises more confusion than the potential race condition it might mitigate. I think the right approach would be to return a default setting when the file cannot be opened after the retry, or when the file is corrupted (schema validation would be good but that would impact the startup time).

The current design is to only read the particular setting that is requested, which will result in opening the files multiple times but actually reading the whole file every time. I don't think this is the right thing to do. We probably should be very explicit that the setting file is read only one time at startup, and we process all settings that is present in the file. In this way, the file is read only once, which means less race conditions, and less file access cost.

Reply:
Should I change anything in the PR?

Reply:
Yes, I think changing `FileShare.ReadWrite` to `FileShare.Read` :)

Reply:
I am not sure that this resolve the race condition in our xUnit tests - I started the PR to get this.

Reply:
@daxian-dbw 	returning a default settings file could result in security settings in the config file being bypassed.

Reply:
> Your xUnit tests actively modify the config file (write lock?) and this lead to crash other tests .

@iSazonov that's unfortunate to hear :(  Then let's go with `FileShare.ReadWrite` for now.

@TravisEz13 I agree. It sounds to me crashing pwsh when config file parsing fails is more securer :)

Reply:
@daxian-dbw @TravisEz13 Could you please add more feedback about "security settings" to my RFC? We could discuss this there. I don't like the "crashing pwsh" because it could be a `DoS attack even by admin user`.